### ISSUE TITLE
.sb loading support

### ISSUE THREAD

**Issue by adroitwhiz** [PullJosh/sb-edit#37]
The original Scratch 1.x file format (`.sb`) would be good to implement.

The following sources might be useful for figuring out how it works:
- [The Scratch Project File Format documentation from John Maloney, Scratch 1.x developer.](https://download.scratch.mit.edu/source-code/ScratchFileFormat.txt) This is out-of-date with regards to newer Scratch features (e.g. lists + comments) but could be a good starting point.
- [`kurt`](https://github.com/tjvr/kurt), a Python library which supports reading the Scratch 1.x file format.
- [`scratch-sb1-converter`](https://github.com/llk/scratch-sb1-converter), the JS library that Scratch itself currently uses to convert .sb files into .sb2.
- [`ObjReader.as`](https://github.com/LLK/scratch-flash/blob/646523e6846ad0dd993213a38b46fe2ea511d026/src/util/ObjReader.as), the ActionScript code used by Scratch 2.0 to read .sb files.

Scratch 1.x uses mostly the same block opcodes as 2.0, so the same opcode mappings could potentially be used for both formats. As such, this is kinda blocked on a .sb2 implementation.

**Comment by easrng** [comment]
This can be implemented using sb2 loading and https://github.com/LLK/scratch-sb1-converter

**Comment by BoomerScratch** [comment]
[scratch-sb1-converter-bundle.zip](https://github.com/PullJosh/sb-edit/files/4889780/scratch-sb1-converter-bundle.zip)
We can use the JS file I have above. Then you can use `ScratchSB1Converter`. If you don't know how to use it, here is an example:
```javascript
fetch("https://projects.scratch.mit.edu/1000").then(res => res.arrayBuffer()).then(data => new ScratchSB1Converter.SB1File(data)).then(data => {
  console.log(data.json); // This logs a Scratch 1 project as a Scratch 2 JSON object in the console.
  console.log(data.zip); // This logs all bytes of the Scratch 1 project's assets in the console (but not the JSON).
});

**Comment by easrng** [comment]
Yeah, we know about that. Currently it is blocked by sb2 loading support.

**Comment by adroitwhiz** [comment]
I'm interested on working this after the library is recoded to fix #100. I was blocked in the past on a resolution to #53, but I think we should be good once #100 is fixed.

**Comment by towerofnix** [comment]
Sounds good, that makes sense!
